### PR TITLE
Rename URL API to URI

### PR DESCRIPTION
**Pull request by tobrun**
Follow up from https://github.com/mapbox/mapbox-gl-native/pull/14750. 
Closes https://github.com/mapbox/mapbox-gl-native/issues/14811.

### CODE REVIEW

**Comment by tobrun** [comment]
Renamed the PR to reflect changes made, this PR also addresses MapboxMapOptions and any kind of BaseUrl configuration.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java [hunk 1]
```diff
@@ -660,9 +683,20 @@ public boolean getRenderSurfaceOnTop() {
...
+   * @deprecated use {@link #getApiBaseUri()} instead
    */
+  @Deprecated
   public String getApiBaseUrl() {
-    return apiBaseUrl;
+    return apiBaseUri;
+  }
+
+  /**
+   * Get the current configured API endpoint base URL.
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java]: URI

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java [hunk 2]
```diff
@@ -660,9 +683,20 @@ public boolean getRenderSurfaceOnTop() {
...
+   *
+   * @return Base URL to be used API endpoint.
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java]: URI

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Style.java [hunk 3]
```diff
@@ -59,11 +59,24 @@ private Style(@NonNull Builder builder, @NonNull NativeMap nativeMap) {
...
    *
    * @return the style url
+   * @deprecated use {@link #getUri()} instead
    */
   @NonNull
+  @Deprecated
   public String getUrl() {
-    validateState("getUrl");
-    return nativeMap.getStyleUrl();
+    validateState("getUri");
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Style.java]: getUrl

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java [hunk 4]
```diff
@@ -236,9 +250,18 @@ public LatLngBounds getRegion() {
...
+     * @deprecated use {@link #getStyleUri()} instead
      */
+    @Deprecated
     public String getStyleUrl() {
-      return styleUrl;
+      return styleUri;
+    }
+
+    /**
+     * @return the style url
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java]: uri

**Pull request merged**